lib/inspec/reporters.rb

```diff
@@ -30,7 +30,10 @@ def self.render(reporter, run_data)
     when "yaml"
       reporter = Inspec::Reporters::Yaml.new(config)
     else
-      raise NotImplementedError, "'#{name}' is not a valid reporter type."
+      # If we made it here, it must be a plugin, and we know it exists (because we validated it in config.rb)
+      activator = Inspec::Plugin::V2::Registry.instance.find_activator(plugin_type: :reporter, activator_name: name.to_sym)
```

Comment:
A plugin naming collision will throw an error if there is a non-unique name loaded. So if a user installs an `inspec-my-yaml-reporter` that implements a `reporter: yaml` activator (and we've migrated our own YAML reporter), then this line will throw an exception because it expects to find a unique result.

Reply:
Cool, thanks. It seems fair that we require users to rename their plugin (or move over to using the one we ship, if they do the same thing) :+1: